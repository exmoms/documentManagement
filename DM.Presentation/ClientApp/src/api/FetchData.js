/**
 * @file this file includes functions that fetch data from API
 */

// Generic Delete Method
export async function deleteFromApi(url) {
  let res = await fetch(url, {
    method: "DELETE",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  });
  return res;
}

// Get MetaData Models
export async function fetchData(url) {
  let object = {};
  /*
   * send the request through a CORS proxy
   * https://stackoverflow.com/questions/43262121/trying-to-use-fetch-and-pass-in-mode-no-cors/43268098#43268098
   */
  await fetch(url, {
    method: "GET",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  })
    .then((response) => response.json())
    .then(
      (data) => (object = data),
      (error) => console.log(error)
    );
  return object;
}
