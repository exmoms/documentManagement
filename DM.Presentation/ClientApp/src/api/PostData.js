/**
 * @file This file includes functions that help to post data to a specific API
 * @version 1.0
 */

/**
 * Post new document data.
 * Upload form-data, scanned copy of doc and attachments.
 * @author Youssef Shaaban <y.shaaban@lit-co.net>
 * @param {*} documentData
 * @param {*} images
 * @param {*} attachments
 * @param {*} url
 * @param {*} option its values either "Add" for add new document version or "Update" to update exsiting document Version
 */
export async function postDocumentData(
  documentData,
  scannedPages,
  attachments,
  url = "",
  option
) {
  var formData = new FormData();

  if (process.env.NODE_ENV === "development") {
    console.log(documentData);
    console.log(attachments);
    console.log(scannedPages);
  }

  switch (option) {
    case "Add":
      formData.append("Document", JSON.stringify(documentData));
      break;
    case "Update":
      formData.append("Version", JSON.stringify(documentData));
      break;
    default:
  }

  attachments.forEach((attachment) => {
    formData.append("Attachments", attachment);
  });
  scannedPages.forEach((scannedPage) => {
    formData.append("Scans", scannedPage);
  });

  let response = {};
  await fetch(url, {
    // content-type header should not be specified!
    method: "POST",
    headers: {
      Accept: "application/json",
      // "Content-Type": "application/json",
    },
    body: formData,
  }).then(
    (res) => {
      response = res;
      // response.json();
    },
    (error) => console.log("[ERROR]:", error)
  );
  return response;
}

/**
 * Post Attachment Data.
 * Upload attachment json data + attachment file  .
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @param {*} payload
 * @param {*} attachmentFile
 * @param {*} url
 */
export async function postAttachmentData(payload, attachmentFile, url = "") {
  var formData = new FormData();

  formData.append("Attachment", JSON.stringify(payload));
  formData.append("FileAttachment", attachmentFile);

  let response = {};
  await fetch(url, {
    // content-type header should not be specified!
    method: "POST",
    body: formData,
  }).then(
    (res) => {
      response = res;
      // response.json();
    },
    (error) => console.log("[ERROR]:", error)
  );
  return response;
}

/**
 * Post Request To API
 * Send post request to api with  payload  and return a response.
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @param {*} url The api url
 * @param {*} payload js object which will be sent as request body.
 */
export async function postDataToAPI(url, payload = null, method = "POST") {

  const response = await fetch(url, {
    method: method,
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(payload),
  });
  return response;
}
