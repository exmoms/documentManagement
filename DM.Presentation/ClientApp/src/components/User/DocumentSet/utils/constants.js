const FILE = 0;
const FOLDER = 1;

export { FILE, FOLDER };
