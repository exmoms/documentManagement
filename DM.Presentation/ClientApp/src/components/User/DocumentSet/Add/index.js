/**
 * @file Contains Add component implementaion
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @version 1.0
 */

import React, { Fragment, Component } from "react";
import styled from "styled-components";
import AddToSetDialog from "./AddToSetDialog";
import AddDocumentToSet from "../AddDocumentToSet";
import AddExistingSetToSet from "../AddExistingSetToSet";
import CreateDocumentSet from "../CreateDocumentSet";
import { FOLDER, FILE } from "../utils/constants";

class Add extends Component {
  constructor(props) {
    super(props);

    this.state = {
      open: false,
      showAddDocument: false,
      showAddNewSet: false,
      showAddExistingSet: false,
    };
  }

  handleOpen = () => {
    this.setState({
      open: true,
      showAddDocument: false,
      showAddNewSet: false,
      showAddExistingSet: false,
    });
  };

  handleChoice = (type, isNew) => {
    if (type === FOLDER) {
      if (isNew) {
        this.setState({
          open: false,
          showAddNewSet: true,
          showAddDocument: false,
          showAddExistingSet: false,
        });
      } else {
        this.setState({
          open: false,
          showAddNewSet: false,
          showAddDocument: false,
          showAddExistingSet: true,
        });
      }
    } else if (type === FILE) {
      this.setState({
        open: false,
        showAddDocument: true,
        showAddNewSet: false,
      });
    } else {
      this.setState({
        open: false,
        showAddDocument: false,
        showAddNewSet: false,
        showAddExistingSet: false,
      });
    }
  };

  closeAddDocument = () => {
    this.setState({
      showAddDocument: false,
    });
  };

  closeAddNewSet = () => {
    this.setState({
      showAddNewSet: false,
    });
  };

  closeAddExistingSet = () => {
    this.setState({
      showAddExistingSet: false,
    });
  };

  render() {
    return (
      <Fragment>
        <Container onClick={() => this.handleOpen()}>+</Container>
        <AddToSetDialog
          set={this.props.set}
          show={this.state.open}
          action={this.handleChoice}
        />

        {this.state.showAddDocument && (
          <AddDocumentToSet
            set={this.props.set}
            show={this.state.showAddDocument}
            action={this.closeAddDocument}
          />
        )}

        {this.state.showAddExistingSet && (
          <AddExistingSetToSet
            set={this.props.set}
            show={this.state.showAddExistingSet}
            action={this.closeAddExistingSet}
          />
        )}

        {this.state.showAddNewSet && (
          <CreateDocumentSet
            set={this.props.set}
            show={this.state.showAddNewSet}
            action={this.closeAddNewSet}
          />
        )}
      </Fragment>
    );
  }
}

export default Add;

const Container = styled.div`
  height: 109px;
  width: 96px;
  border: 3px dashed #dee0e4;
  display: flex;
  justify-content: center;
  border-radius: 8px;
  align-items: center;
  font-size: 30px;
  color: #dee0e4;
  margin: -6px 21px;
  cursor: copy;
`;
