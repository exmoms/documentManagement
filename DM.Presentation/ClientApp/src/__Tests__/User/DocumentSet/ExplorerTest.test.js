import React from "react";
import ReactDom from "react-dom";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { mount } from "enzyme";
import { createMount } from '@material-ui/core/test-utils';
import Dialog from "@material-ui/core/Dialog";
import { Fragment } from "react";
import ListItem from "@material-ui/core/ListItem";
import Explorer from "../../../components/User/DocumentSet/Explorer";
import SearchDocuments from "../../../components/User/SearchDocuments";
import DocumentTable from "../../../components/User/DocumentTable";
import Button from "@material-ui/core/Button";
import * as FetchMock from "../../../api/FetchData";
import ListItemText from "@material-ui/core/ListItemText";
import * as PostDataMock from "../../../api/PostData";
import List from "@material-ui/core/List";
import { BrowserRouter as Router } from "react-router-dom";
import { Redirect } from "react-router";
import { Typography } from "@material-ui/core";
import DialogContent from "@material-ui/core/DialogContent";
import { TextField } from "@material-ui/core";
import Icon from "../../../components/User/DocumentSet/Icon";
import Add from "../../../components/User/DocumentSet/Add";
import { createBrowserHistory } from "history";
import * as BrowserHistoryMock from "history";
import { FILE, FOLDER } from "../../../components/User/DocumentSet/utils/constants";
afterEach(cleanup);

describe("Explorer", () => {
    afterEach(() => {
        jest.clearAllMocks();
      });
    test("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDom.render(<Explorer />, div);
      });

    test("Explorer matches snapshot", () => {
    const renderedValue = createMount()(
        <Explorer />
      );
      expect(renderedValue.html()).toMatchSnapshot();
    });
  //   test("refreshDocumentSet functionality when set.id =0",  () =>{
  //     const mockSuccessResponseMetaDataModel = [
  //         {id:1, addedDate:"addedDate", modifiedDate:"modifiedDate",name:"name",attachedDocuments:[{id:1,name:"attched 1"}],childrenDocumentSets:[{id:1,name:"children 1"}]
  //     ,documentName:"documentName" , metadataModelId:2,metadataModelName:"metadataModelName",deletedDate:"deletedDate",latestVersion:2,documentVersion:1,attachments:"attachments"}
  //       ];
  //       const mockJsonPromiseMetaDataModel = Promise.resolve(
  //         mockSuccessResponseMetaDataModel
  //       ); // 2
  //       const fetchingMetaDataModel = jest
  //         .spyOn(FetchMock, "fetchData")
  //         .mockImplementation(() => mockJsonPromiseMetaDataModel);
  //         let  wrapper;
  //         return mockJsonPromiseMetaDataModel.then(() => {
  //       wrapper = mount(
  //             <Explorer/>
  //           );
  //         }).then(() => {
  //             wrapper.update();
  //             wrapper.render();
  //             expect(fetchingMetaDataModel).toHaveBeenCalledTimes(1);
  //             expect(fetchingMetaDataModel.mock.calls[0][0]).toBe('/api/DocumentSet/GetRoots');
  //             expect(wrapper.find(Icon).length).toBe(1);
  //             expect(wrapper.find(Icon).at(0).props().type).toBe(FOLDER);
  //             expect(wrapper.find(Add).length).toBe(1);
  //             expect(wrapper.find(Add).at(0).props().set).toStrictEqual({"id": 0, "name": "Home"});
  //              expect(wrapper.find("#notification").length).toBe(1);
  //             expect(wrapper.find("#notification").at(0).props().open).toBeFalsy(); 
  //             expect(wrapper.find("#notification").at(0).props().error).toBeFalsy(); 
  //             expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual([]); 
  //         act(() => {
  //           wrapper
  //             .find(Icon)
  //             .at(0)
  //             .props()
  //             .refreshSet({id:0});
  //         });
  //         wrapper.update();
  //         wrapper.render();
  //         expect(fetchingMetaDataModel).toHaveBeenCalledTimes(2);
  //         expect(fetchingMetaDataModel.mock.calls[1][0]).toBe('/api/DocumentSet/GetRoots');
  //         expect(wrapper.find(Icon).length).toBe(1);
  //         expect(wrapper.find(Icon).at(0).props().type).toBe(FOLDER);
  //         expect(wrapper.find(Icon).at(0).props().set).toStrictEqual({ id: 0, name: "Home" });
  //         expect(wrapper.find("#notification").at(0).props().open).toBeFalsy(); 
  //         expect(wrapper.find("#notification").at(0).props().error).toBeFalsy(); 
  //         expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual( []); 
  //         expect(wrapper.find("#goToUpperLevelId").length).toBe(1); 
  //       });
  // })
  

//   test("refreshDocumentSet functionality when set.id=1",  () =>{
//     const mockSuccessResponseMetaDataModel = [
//       {id:1, addedDate:"addedDate", modifiedDate:"modifiedDate",name:"name",attachedDocuments:[{id:1,name:"attched 1"}],childrenDocumentSets:[{id:1,name:"children 1"}]
//   ,documentName:"documentName" , metadataModelId:2,metadataModelName:"metadataModelName",deletedDate:"deletedDate",latestVersion:2,documentVersion:1,attachments:"attachments"}
//     ];
//     const mockJsonPromiseMetaDataModel = Promise.resolve(
//       mockSuccessResponseMetaDataModel
//     ); // 2
//     const mockSuccessResponseMetaDataModel2 = {attachedDocuments:[{id:1,name:"attched 1"}],childrenDocumentSets:[{id:1,name:"children 1"}]};
//         const mockJsonPromiseMetaDataModel2 = Promise.resolve(
//           mockSuccessResponseMetaDataModel2
//         ); // 2
//       const fetchingMetaDataModel = jest
//         .spyOn(FetchMock, "fetchData")
//         .mockReturnValueOnce(mockJsonPromiseMetaDataModel).mockReturnValueOnce(mockJsonPromiseMetaDataModel2);
//         let  wrapper;
//         return mockJsonPromiseMetaDataModel.then(() => {
//       wrapper = mount(
//             <Explorer/>
//           );
//         }).then(() => {
//             wrapper.update();
//             wrapper.render();
//             expect(fetchingMetaDataModel).toHaveBeenCalledTimes(1);
//             expect(fetchingMetaDataModel.mock.calls[0][0]).toBe('/api/DocumentSet/GetRoots');
//             expect(wrapper.find(Icon).length).toBe(1);
//             expect(wrapper.find(Icon).at(0).props().type).toBe(FOLDER);
//             expect(wrapper.find(Add).length).toBe(1);
//             expect(wrapper.find(Add).at(0).props().set).toStrictEqual({"id": 0, "name": "Home"});
//              expect(wrapper.find("#notification").length).toBe(1);
//             expect(wrapper.find("#notification").at(0).props().open).toBeFalsy(); 
//             expect(wrapper.find("#notification").at(0).props().error).toBeFalsy(); 
//             expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual([]); 
//             return mockJsonPromiseMetaDataModel2.then(() => {
//         act(() => {
//           wrapper
//             .find(Icon)
//             .at(0)
//             .props()
//             .refreshSet({id:1});
//         });
//       }).then(() => {
//         wrapper.update();
//         wrapper.render();
//         expect(fetchingMetaDataModel).toHaveBeenCalledTimes(2);
//         expect(fetchingMetaDataModel.mock.calls[1][0]).toBe('/api/DocumentSet/GetDocumentSetByID?document_set_id=1');
//         expect(wrapper.find(Icon).length).toBe(2);
//         expect(wrapper.find(Icon).at(0).props().type).toBe(FOLDER);
//         expect(wrapper.find("#notification").at(0).props().open).toBeTruthy(); 
//         expect(wrapper.find("#notification").at(0).props().error).toBeFalsy(); 
//         expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual( []); 
//         expect(wrapper.find("#goToUpperLevelId").length).toBe(1); 
//       });
//     });
// })


  //   test("componentDidUpdate functionality when (state.id !== 0) + refreshDocumentSet when set.id = 1",  () =>{
  //     jest.clearAllMocks();
  //     const mockSuccessResponseMetaDataModel1 = [
  //         {id:1, addedDate:"addedDate", modifiedDate:"modifiedDate",name:"name",attachedDocuments:[{id:1,name:"attched 1"}],childrenDocumentSets:[{id:1,name:"children 1"}]
  //     ,documentName:"documentName" , metadataModelId:2,metadataModelName:"metadataModelName",deletedDate:"deletedDate",latestVersion:2,documentVersion:1,attachments:"attachments"}
  //       ];
  //       const mockSuccessResponseMetaDataModel2 = {attachedDocuments:[{id:1,name:"attched 1"}],childrenDocumentSets:[{id:1,name:"children 1"}]};
  //       const mockJsonPromiseMetaDataModel1 = Promise.resolve(
  //         mockSuccessResponseMetaDataModel1
  //       ); // 2
  //       const mockJsonPromiseMetaDataModel2 = Promise.resolve(
  //         mockSuccessResponseMetaDataModel2
  //       ); // 2
  //       let replaceMock = jest.fn();
  //       let createBrowserHistoryReturnedValue={location:{state:{id:1 , name :"Name A" , openNotifiction:true, error:true,errorMessage:["errorMessage"]}} , replace:()=>{ replaceMock }};
  //          const createBrowserHistoryMock=  jest.spyOn(BrowserHistoryMock, "createBrowserHistory").mockReturnValueOnce(createBrowserHistoryReturnedValue).mockReturnValueOnce(createBrowserHistoryReturnedValue).mockReturnValueOnce(createBrowserHistoryReturnedValue);
  //       const fetchingMetaDataModel = jest
  //         .spyOn(FetchMock, "fetchData")
  //         .mockReturnValueOnce(mockJsonPromiseMetaDataModel1).mockReturnValueOnce(mockJsonPromiseMetaDataModel2).mockReturnValueOnce(mockJsonPromiseMetaDataModel2).mockReturnValueOnce(mockJsonPromiseMetaDataModel2);
  //         let  wrapper;
  //         return mockJsonPromiseMetaDataModel1.then(() => {
  //       wrapper = mount(
  //             <Explorer/>
  //           );
  //         }).then(() => {
  //           return mockJsonPromiseMetaDataModel2.then(() => {
  //             wrapper.update();
  //           }).then(() => {
  //             wrapper.update();
  //             wrapper.render();
  //              expect(createBrowserHistoryMock).toHaveBeenCalledTimes(3);
  //             expect(fetchingMetaDataModel).toHaveBeenCalledTimes(4);
  //             // expect(replaceMock).toHaveBeenCalledTimes(1);
  //             expect(fetchingMetaDataModel.mock.calls[0][0]).toBe('/api/DocumentSet/GetRoots');
  //             expect(fetchingMetaDataModel.mock.calls[1][0]).toBe('/api/DocumentSet/GetDocumentSetByID?document_set_id=1');
  //             expect(wrapper.find(Icon).length).toBe(2);
  //             expect(wrapper.find(Icon).at(0).props().type).toBe(FOLDER);
  //             expect(wrapper.find(Icon).at(0).props().set).toStrictEqual({"id": 1, "name": "Name A"});
  //             expect(wrapper.find(Icon).at(1).props().type).toBe(FILE);
  //             expect(wrapper.find(Icon).at(1).props().set).toStrictEqual({"id": 1, "name": "Name A"});
  //             expect(wrapper.find("#notification").at(0).props().open).toBeTruthy(); 
  //             expect(wrapper.find("#notification").at(0).props().error).toBeTruthy(); 
  //             expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual(["errorMessage"]); 
  //             expect(wrapper.find("#notification").length).toBe(1);
  //         });
  //       });
  // })
    // test("componentDidUpdate functionality when (state.id === 0)",  () =>{
    //     const mockSuccessResponseMetaDataModel = [
    //         {id:1, addedDate:"addedDate", modifiedDate:"modifiedDate",name:"name",attachedDocuments:[{id:1,name:"attched 1"}],childrenDocumentSets:[{id:1,name:"children 1"}]
    //     ,documentName:"documentName" , metadataModelId:2,metadataModelName:"metadataModelName",deletedDate:"deletedDate",latestVersion:2,documentVersion:1,attachments:"attachments"}
    //       ];
    //       const mockJsonPromiseMetaDataModel = Promise.resolve(
    //         mockSuccessResponseMetaDataModel
    //       ); // 2
    //       let replaceMock = jest.fn();
    //       let createBrowserHistoryReturnedValue={location:{state:{id:0 , name :"Name A" , openNotifiction:false, error:false,errorMessage:[]}} , replace:({state})=>{ replaceMock({state}) }};
    //          const createBrowserHistoryMock=  jest.spyOn(BrowserHistoryMock, "createBrowserHistory").mockReturnValueOnce(createBrowserHistoryReturnedValue);
    //       const fetchingMetaDataModel = jest
    //         .spyOn(FetchMock, "fetchData")
    //         .mockImplementation(() => mockJsonPromiseMetaDataModel);
    //         let  wrapper;
    //         return mockJsonPromiseMetaDataModel.then(() => {
    //       wrapper = mount(
    //             <Explorer/>
    //           );
    //         }).then(() => {
    //             wrapper.update();
    //             wrapper.render();
    //             expect(createBrowserHistoryMock).toHaveBeenCalledTimes(1);
    //             expect(fetchingMetaDataModel).toHaveBeenCalledTimes(2);
    //             // expect(replaceMock).toHaveBeenCalledTimes(1);
    //             expect(fetchingMetaDataModel.mock.calls[0][0]).toBe('/api/DocumentSet/GetRoots');
    //             expect(fetchingMetaDataModel.mock.calls[1][0]).toBe('/api/DocumentSet/GetRoots');
    //             expect(wrapper.find(Icon).length).toBe(1);
    //             expect(wrapper.find(Icon).at(0).props().type).toBe(FOLDER);
    //             expect(wrapper.find(Add).length).toBe(1);
    //             expect(wrapper.find(Add).at(0).props().set).toStrictEqual({"id": 0, "name": "Home"});
    //              expect(wrapper.find("#notification").length).toBe(1);
    //             expect(wrapper.find("#notification").at(0).props().open).toBeFalsy(); 
    //             expect(wrapper.find("#notification").at(0).props().error).toBeFalsy(); 
    //             expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual([]); 
 
    //         });
    // })
   

    // test("showMore functionality + goToUpperLevel + setOpenNotification",  () =>{
    //     const mockSuccessResponseMetaDataModel1 = [
    //         {id:1, addedDate:"addedDate", modifiedDate:"modifiedDate",name:"name",attachedDocuments:[{id:1,name:"attched 1"}],childrenDocumentSets:[{id:1,name:"children 1"}]
    //     ,documentName:"documentName" , metadataModelId:2,metadataModelName:"metadataModelName",deletedDate:"deletedDate",latestVersion:2,documentVersion:1,attachments:"attachments"}
    //       ];
    //       const mockSuccessResponseMetaDataModel2 = {attachedDocuments:[],childrenDocumentSets:[{id:1,name:"children 1"}]};
    //       const mockJsonPromiseMetaDataModel1 = Promise.resolve(
    //         mockSuccessResponseMetaDataModel1
    //       ); // 2
    //       const mockJsonPromiseMetaDataModel2 = Promise.resolve(
    //         mockSuccessResponseMetaDataModel2
    //       ); // 2
    //       const fetchingMetaDataModel = jest
    //         .spyOn(FetchMock, "fetchData")
    //         .mockReturnValueOnce(mockJsonPromiseMetaDataModel1).mockReturnValueOnce(mockJsonPromiseMetaDataModel2);
    //         let  wrapper;
    //         return mockJsonPromiseMetaDataModel1.then(() => {
    //       wrapper = mount(
    //             <Explorer/>
    //           );
    //         }).then(() => {
    //             wrapper.update();
    //             wrapper.render();
    //             expect(fetchingMetaDataModel).toHaveBeenCalledTimes(1);
    //             // expect(replaceMock).toHaveBeenCalledTimes(1);
    //             expect(fetchingMetaDataModel.mock.calls[0][0]).toBe('/api/DocumentSet/GetRoots');
    //             expect(wrapper.find(Icon).length).toBe(1);
    //             expect(wrapper.find(Icon).at(0).props().type).toBe(FOLDER);
    //          //calling showMore
    //          return mockJsonPromiseMetaDataModel2.then(() => {
    //          act(() => {
    //             wrapper
    //               .find(Icon)
    //               .at(0)
    //               .props()
    //               .enterFolder(1,"tested Value");
    //           });
    //         }).then(() => {
    //         wrapper.update();
    //         wrapper.render();
    //         expect(fetchingMetaDataModel).toHaveBeenCalledTimes(2);
    //         expect(fetchingMetaDataModel.mock.calls[1][0]).toBe('/api/DocumentSet/GetDocumentSetByID?document_set_id=1');
    //         expect(wrapper.find(Icon).length).toBe(1);
    //         expect(wrapper.find(Icon).at(0).props().type).toBe(FOLDER);
    //         expect(wrapper.find(Icon).at(0).props().set).toStrictEqual({"id": 1, "name": "tested Value"});
    //         expect(wrapper.find("#notification").at(0).props().open).toBeFalsy(); 
    //         expect(wrapper.find("#notification").at(0).props().error).toBeFalsy(); 
    //         expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual([]); 
    //         expect(wrapper.find("#goToUpperLevelId").length).toBe(2);
    //         act(() => {
    //           wrapper
    //             .find("#goToUpperLevelId")
    //             .at(0)
    //             .props()
    //             .onClick();
    //         });
    //         wrapper.update();
    //         wrapper.render();
    //         expect(fetchingMetaDataModel).toHaveBeenCalledTimes(3);
    //         expect(fetchingMetaDataModel.mock.calls[2][0]).toBe('/api/DocumentSet/GetRoots');
    //         expect(wrapper.find(Icon).length).toBe(1);
    //         expect(wrapper.find(Icon).at(0).props().type).toBe(FOLDER);
    //         expect(wrapper.find(Icon).at(0).props().set).toStrictEqual({"id": 1, "name": "tested Value"});
    //         expect(wrapper.find("#notification").at(0).props().open).toBeFalsy(); 
    //         expect(wrapper.find("#notification").at(0).props().error).toBeFalsy(); 
    //         expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual([]); 
    //         expect(wrapper.find("#goToUpperLevelId").length).toBe(2); 

    //         act(() => {
    //           wrapper
    //             .find("#notification")
    //             .at(0)
    //             .props()
    //             .setOpen(true);
    //         });
    //         wrapper.update();
    //         wrapper.render();
    //         expect(wrapper.find("#notification").at(0).props().open).toBeTruthy(); 

    //         act(() => {
    //           wrapper
    //             .find(Icon)
    //             .at(0)
    //             .props()
    //             .showNotification(false , true , ["erroMessage"]);
    //         });
    //         wrapper.update();
    //         wrapper.render();
    //         expect(wrapper.find("#notification").at(0).props().open).toBeFalsy(); 
    //         expect(wrapper.find("#notification").at(0).props().error).toBeTruthy(); 
    //         expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual(["erroMessage"]); 

          
            
    //       });
    //       });     
          
    // })
    
  //   test("showMore functionality + goToUpperLevel when index >= 1",  () =>{
  //     const mockSuccessResponseMetaDataModel1 = [
  //         {id:1, addedDate:"addedDate", modifiedDate:"modifiedDate",name:"name",attachedDocuments:[{id:1,name:"attched 1"}],childrenDocumentSets:[{id:1,name:"children 1"}]
  //     ,documentName:"documentName" , metadataModelId:2,metadataModelName:"metadataModelName",deletedDate:"deletedDate",latestVersion:2,documentVersion:1,attachments:"attachments"}
  //       ];
  //       const mockSuccessResponseMetaDataModel2 = {attachedDocuments:[],childrenDocumentSets:[{id:1,name:"children 1"}]};
  //       const mockJsonPromiseMetaDataModel1 = Promise.resolve(
  //         mockSuccessResponseMetaDataModel1
  //       ); // 2
  //       const mockJsonPromiseMetaDataModel2 = Promise.resolve(
  //         mockSuccessResponseMetaDataModel2
  //       ); // 2
  //       const fetchingMetaDataModel = jest
  //         .spyOn(FetchMock, "fetchData")
  //         .mockReturnValueOnce(mockJsonPromiseMetaDataModel1).mockReturnValueOnce(mockJsonPromiseMetaDataModel2).mockReturnValueOnce(mockJsonPromiseMetaDataModel2).mockReturnValueOnce(mockJsonPromiseMetaDataModel2).mockReturnValueOnce(mockJsonPromiseMetaDataModel2);
  //         let  wrapper;
  //         return mockJsonPromiseMetaDataModel1.then(() => {
  //       wrapper = mount(
  //             <Explorer/>
  //           );
  //         }).then(() => {
  //             wrapper.update();
  //             wrapper.render();
  //             expect(fetchingMetaDataModel).toHaveBeenCalledTimes(1);
  //             // expect(replaceMock).toHaveBeenCalledTimes(1);
  //             expect(fetchingMetaDataModel.mock.calls[0][0]).toBe('/api/DocumentSet/GetRoots');
  //             expect(wrapper.find(Icon).length).toBe(1);
  //             expect(wrapper.find(Icon).at(0).props().type).toBe(FOLDER);
  //          //calling showMore
  //          return mockJsonPromiseMetaDataModel2.then(() => {
  //          act(() => {
  //             wrapper
  //               .find(Icon)
  //               .at(0)
  //               .props()
  //               .enterFolder(1,"tested Value1");
  //           });
  //           wrapper.update();
  //           act(() => {
  //             wrapper
  //               .find(Icon)
  //               .at(0)
  //               .props()
  //               .enterFolder(2,"tested Value2");
  //           });
  //         }).then(() => {
  //         wrapper.update();
  //         wrapper.render();
  //         expect(fetchingMetaDataModel).toHaveBeenCalledTimes(3);
  //         expect(fetchingMetaDataModel.mock.calls[2][0]).toBe('/api/DocumentSet/GetDocumentSetByID?document_set_id=2');
  //         expect(wrapper.find(Icon).length).toBe(1);
  //         expect(wrapper.find(Icon).at(0).props().type).toBe(FOLDER);
  //         expect(wrapper.find(Icon).at(0).props().set).toStrictEqual({"id": 2, "name": "tested Value2"});
  //         expect(wrapper.find("#notification").at(0).props().open).toBeFalsy(); 
  //         expect(wrapper.find("#notification").at(0).props().error).toBeFalsy(); 
  //         expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual([]); 
  //         expect(wrapper.find("#goToUpperLevelId").length).toBe(3);
  //         act(() => {
  //           wrapper
  //             .find("#goToUpperLevelId")
  //             .at(1)
  //             .props()
  //             .onClick();
  //         });
  //         wrapper.update();
  //         wrapper.render();
  //         expect(fetchingMetaDataModel).toHaveBeenCalledTimes(4);
  //         expect(fetchingMetaDataModel.mock.calls[3][0]).toBe('/api/DocumentSet/GetDocumentSetByID?document_set_id=1');
  //         expect(wrapper.find(Icon).length).toBe(1);
  //         expect(wrapper.find(Icon).at(0).props().type).toBe(FOLDER);
  //         expect(wrapper.find(Icon).at(0).props().set).toStrictEqual({"id": 2, "name": "tested Value2"});
  //         expect(wrapper.find("#notification").at(0).props().open).toBeFalsy(); 
  //         expect(wrapper.find("#notification").at(0).props().error).toBeFalsy(); 
  //         expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual([]); 
  //         expect(wrapper.find("#goToUpperLevelId").length).toBe(3);   
      
  //       });     
  //     });      
  // })
});