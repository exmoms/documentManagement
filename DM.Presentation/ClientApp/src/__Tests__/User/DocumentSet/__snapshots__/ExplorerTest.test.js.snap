// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explorer Explorer matches snapshot 1`] = `"<div class=\\"sc-fzplWN haKEer\\"><span id=\\"goToUpperLevelId\\">path_Home  / </span></div><div class=\\"sc-fzpans cIyyYC\\"><div class=\\"sc-AxhUy feYMlc\\">+</div></div>"`;
